contracts/FIFOMarket.sol

```diff
@@ -253,4 +339,72 @@ contract FIFOMarket is
       AccessControlEnumerableUpgradeable.supportsInterface(interfaceId) ||
       ERC1155ReceiverUpgradeable.supportsInterface(interfaceId);
   }
+
+  function _incrementCurrentSupplierAddress() private {
+    // Update the current supplier to be the next of the current supplier
+    _currentSupplierAddress = _suppliersInRoundRobinOrder[
+      _currentSupplierAddress
+    ].nextSupplierAddress;
+  }
+
+  function _addActiveSupplier(address supplierAddress) private {
+    // If this is the first supplier to be added, update the intialized addresses.
+    if (_currentSupplierAddress == address(0)) {
+      _currentSupplierAddress = supplierAddress;
+      _suppliersInRoundRobinOrder[supplierAddress] = RoundRobinOrder({
+        previousSupplierAddress: supplierAddress,
+        nextSupplierAddress: supplierAddress
+      });
+    } else {
+      // Add the new supplier to the round robin order, with the current supplier as next and the current supplier's previous supplier as previous
+      _suppliersInRoundRobinOrder[supplierAddress] = RoundRobinOrder({
+        previousSupplierAddress: _suppliersInRoundRobinOrder[
+          _currentSupplierAddress
+        ].previousSupplierAddress,
+        nextSupplierAddress: _currentSupplierAddress
+      });
+      // Update the previous supplier to point to the new supplier as next
+      _suppliersInRoundRobinOrder[
+        _suppliersInRoundRobinOrder[_currentSupplierAddress]
+          .previousSupplierAddress
+      ].nextSupplierAddress = supplierAddress;
+      // Update the current supplier to point to the new supplier as previous
+      _suppliersInRoundRobinOrder[_currentSupplierAddress]
+        .previousSupplierAddress = supplierAddress;
```

Comment:
I dont understand how this works-- is there a description somewhere on how this updates state with multiple suppliers?

Reply:
No, I'll go ahead and add some documentation to `FIFOMarket.md`